modules/auxiliary/admin/ldap/vmware_vcenter_vmdir_auth_bypass.rb

```diff
@@ -109,11 +110,13 @@ def run
 
   # This will always return false, since the creds are invalid
   def auth_bypass(ldap)
-    ldap.bind(
-      method: :simple,
-      username: Rex::Text.rand_text_alphanumeric(8..42),
-      password: Rex::Text.rand_text_alphanumeric(8..42)
-    )
+    unless datastore['BIND_DN']
+      ldap.bind(
```

Comment:
when datastore['BIND_DN'] has been provided in options, ldap_connect makes a bind for us.

Reply:
`return if datastore['BIND_DN']` would be a little easier to follow.

Reply:
And you can comment what you just said above it.

Reply:
Corrected. (I though using unless makes you more ruby - where as in Python/C .. I use always if / return :-D )

Reply:
Sure, unless it can be refactored to the "positive" case, especially where a guard clause is suitable. :-)